content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md

```diff
@@ -0,0 +1,254 @@
++++
+title = "Generic Conditional Concurrency Requirement Overview and Design"
+description = "This page provides an overview of `Generic Conditional Concurrency Requirement`"
+weight = 10
+toc = true
+aliases = ["/docs/design/"]
+[menu.main]
+  parent = "Design"
+  weight = 2 
++++
+
+In the genome sequencing flow, sometimes the if-else process of the sequencing step or even the switch-case process needs to be implemented according to the situation. 
+For example, step A is used to judge whether the data is qualified. If the data is qualified, step B is performed, and if it is unqualified, step C is performed.
+
+Kubegene needs generic conditional branching in various scenarios. Kubegene need to dynamic generic conditional branching.
+
+So Kubegene needs if/else conditional branch and switch/case generic conditional branch handling.
+
+# **Generic Conditional Dynamic Concurrency**
+
+The check_result based support is limited for just matching for equality, this generic condition will support the
+different operators like In,Equal,NotIn,Exists etc so this feature is super set of check_result.
+
+The so-called dynamic, that is, the process runs, and decide whether to perform the next step. This is the ability that many other existing processes (such as WDL) do not have.
+Because it is necessary to judge whether the current condition value is true or false according to the "execution result" of another step, 
+Kubegene uses the generic condition feature for dynamically conditional branch judging.
+
+
+switch/case example:
+
+In the following example, the result of job-a is testkey:testvalue, then job-b will execute, and job-c and job-d will not execute.
+
+```yaml
+Workflow:
+  Job-a:
+    tool: nginx:latest
+    commands:
+      - echo testkey:testvalue # <== The execution result is reflected in std_out
+  Job-b:
+    tool: nginx:latest
+    genericcondition:         # <==Judge the result of job-a and decide whether the current step is executed.
+      dependjobname: Job-a    # <==The given one of the matching rule should satisfy the job-a result which should be list of key:value  separated by ,
+      matchrules:
+        - key: testscenario
+          operator: In
+          values:
+           - testwithInOp
+    commands:
+      - echo run-job-b
+  Job-c:
+    cool: nginx:latest
+    genericcondition: 
+       dependjobname: Job-a
+       matchrules:
+        - key: testscenarioinv
+          operator: Exists
+            - testwithInOp
```

Comment:
values ?

Reply:
or should this just for key, and no value ?

Reply:
yes, no values for Exists /DoesNotExists  operators